testplan/runnable.py

```diff
@@ -401,6 +405,16 @@ def _wait_ongoing(self):
                 resource.abort()
 
         while self.active:
+            if self.cfg.timeout and \
+                    time.time() - self._start_time > self.cfg.timeout:
+                self.result.test_report.logger.error(
+                    'Timeout: Aborting execution after {} seconds'.format(
+                        self.cfg.timeout))
+                for dep in self.abort_dependencies():
```

Comment:
Why dont we simply call self.abort() method that is inherited by entity base class?

Reply:
we cannot abort TestRunner, or it won't create test result & generate pdf report. It only aborts executors and the reports of finished multitest can still be fetched.